### PR TITLE
fix(Popup): use original position if none fit in viewport

### LINKED ISSUE #1482: Popup position is not respected when near corner of window
### Steps
1. Create a popup with a trigger at the top right corner of the window.
2. Make it a large popup so it can't fit on any side of the trigger without going out of the window

### Expected Result
It should be positioned according to the position prop.

### Actual Result
It always defaults to the last value in the `POSITIONS` enum excluding the value from the position prop (mostly "bottom center"). And it never uses the user specified value from the prop.

### Version
0.67.0

### Testcase
Just look at https://github.com/Semantic-Org/Semantic-UI-React/blob/master/src/modules/Popup/Popup.js#L234

### PR DESCRIPTION
**Pull request by MindFreeze**
Should fix #1482

### CODE REVIEW

**Comment by MindFreeze** [comment]
I have no idea why the `is always within the viewport` test fails with this change. Some help with this would be appreciated.

File: src/modules/Popup/Popup.js [hunk 1]
```diff
@@ -228,7 +228,7 @@ export default class Popup extends Component {
 
     // Lets detect if the popup is out of the viewport and adjust
     // the position accordingly
-    const positions = _.without(POSITIONS, position)
+    const positions = _.without(POSITIONS, position).push(position)
```
Comment (layershifter) [inline src/modules/Popup/Popup.js]: This line doesn't make sense, you remove `position` from array and then push it again :confused:

Comment (MindFreeze) [reply src/modules/Popup/Popup.js]: I add it to the end of the array, so that if no suitable position can be found, the selected (last) one is used. Otherwise it would just use the last one in POSITIONS, which is `bottom center`.

**Comment by levithomason** [comment]
@MindFreeze any interest in pushing this one over the finish line?  I'd be OK merging it if tests were fixed and a new test added that asserts the new behavior.  Namely, we'd need a test that says if no other position fits in the viewport then the original position is used.

Since this is a very rare edge case, it probably will not get attention from the core team.

**Comment by MindFreeze** [comment]
I'll give it a go

**Comment by MindFreeze** [comment]
@levithomason All done.

**Comment by levithomason** [comment]
You're awesome, thanks much!

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.71.3`
